_docs-sources/iac/usage/using-the-library.md

```diff
@@ -0,0 +1,396 @@
+
+## Create a module
+
+In this section you’ll create a module that provisions an AWS Lambda Function using the [`terraform-aws-lambda`](../../reference/modules/terraform-aws-lambda/lambda/) Gruntwork module. This module automatically creates the AWS IAM role and Cloudwatch Log Group for the Lambda function. For more information about the configuration of the module, refer to the [Library Reference](../../reference/modules/terraform-aws-lambda/lambda/#reference).
+
+### Create the basic file structure
+
+First, create the basic file structure required that will contain the module reference. In this guide, you will create a module named `serverless-api` which contains the reference to the `terraform-aws-lambda` module. This approach is taken to allow you to define a module once, then use it in many environments and regions, while keeping the amount of copy-and-paste work to a minimum.
+
+In this guide, we will use `example` as the name of the environment. In a real-world environment, this might be `dev`, `staging`, `production`, or any other name.
+
+<Tabs groupId="tool-choice">
+<TabItem value="Terraform" label="Terraform" default>
+
+For Terraform, create two paths — one which will contain the reference to the `terraform-aws-lambda` module and one which will contain a reference to the local module (aka ["wrapper module"](#creating-a-wrapper-module-terraform-only)).
+
+```bash
+mkdir -p gw_module_guide/serverless-api/lambda
+touch gw_module_guide/serverless-api/lambda/main.tf
+touch gw_module_guide/serverless-api/lambda/variables.tf
+
+mkdir -p gw_module_guide/example/<YOUR_REGION>
+touch gw_module_guide/example/<YOUR_REGION>/main.tf
+
+mkdir -p gw_module_guide/example/<YOUR_REGION>/src
+touch gw_module_guide/example/<YOUR_REGION>/src/main.py
+```
+
+</TabItem>
+<TabItem value="Terragrunt" label="Terragrunt" default>
+
+For Terragrunt, we recommend that all re-usable infrastructure modules live in a directory called `_envcommon`. Create two paths — one which will contain the reference to the `terraform-aws-lambda` and one which will contain a reference to the local module.
+
+```bash
+mkdir -p gw_module_guide/_envcommon/serverless-api
+touch gw_module_guide/terragrunt.hcl
+touch gw_module_guide/_envcommon/serverless-api/lambda.hcl
+
+mkdir -p gw_module_guide/example/<YOUR REGION>/example/serverless-api
+touch gw_module_guide/example/<YOUR REGION>/example/serverless-api/terragrunt.hcl
+
+mkdir -p gw_module_guide/example/<YOUR REGION>/example/serverless-api/src
+touch gw_module_guide/example/<YOUR REGION>/example/serverless-api/src/main.py
+```
+
+</TabItem>
+</Tabs>
+
+### Create the reference to the Gruntwork module
+
+Next, create the reference to the `terraform-aws-lambda` module...
+
+<Tabs groupId="tool-choice">
+<TabItem value="Terraform" label="Terraform" default>
+
+Define a module block that uses the git url `terraform-aws-lambda` module for the `source` attribute.
```

Comment:
```suggestion
Define a module block that uses the git url of the `terraform-aws-lambda` module for the `source` attribute.
```

Reply:
I did some re-ordering of the wording here, moving the common paragraph up above into the section header. LMK what you think.